enclave_wrangler/datasets.py

```diff
@@ -0,0 +1,355 @@
+
+
+def run(
+    dataset_name: str = None, dataset_rid: str = None, ref: str = 'master', outdir: str = None, outpath: str = None,
+    transforms_only=False
+) -> pd.DataFrame:
+    dataset_rid = FAVORITE_DATASETS[dataset_name]['rid'] if not dataset_rid else dataset_rid
+    dataset_name = FAVORITE_DATASETS_RID_NAME_MAP[dataset_rid] if not dataset_name else dataset_name
+
+    # Download
+    df = pd.DataFrame()
+    if not transforms_only:
+        # TODO: Temp: would be good to accept either 'outdir' or 'outpath'.
+        if not outpath:
+            outpath = os.path.join(outdir, f'{dataset_rid}__{ref}.csv') if outdir else None
+        if os.path.exists(outpath):
+            t = time.ctime(os.path.getmtime(outpath))
+            print(f'Skipping {outpath}: {t}, {os.path.getsize(outpath)} bytes.')
+            return pd.read_csv(outpath)
+        endRef = getTransaction(dataset_rid, ref)
+        args = {'datasetRid': dataset_rid, 'endRef': endRef}
+        file_parts = views2(**args)
+        # asyncio.run(download_and_combine_dataset_parts(datasetRid, file_parts))
+        df: pd.DataFrame = download_and_combine_dataset_parts(dataset_rid, file_parts, outpath=outpath)
+
+    # Transform
+    df2: pd.DataFrame = transform(dataset_name)
+
+    return df2 if df2 else df
+
+
+def run_favorites(outdir: str = CSV_DOWNLOAD_DIR, transforms_only=False):
+    """Run on favorite datasets"""
+    for fav in FAVORITE_DATASETS.values():
+        outpath = os.path.join(outdir, fav['name'] + '.csv')
+        run(dataset_name=fav['name'], outpath=outpath, transforms_only=transforms_only)
+
+
+def get_parser():
+    """Add required fields to parser.
+
+    Returns:
+        ArgumentParser: Argeparse object.
+    """
+    package_description = 'Tool for working w/ the Palantir Foundry enclave API. ' \
+          'This part is for downloading enclave datasets.'
+    parser = ArgumentParser(description=package_description)
+
+    parser.add_argument(
+        '-a', '--auth_token_env_var',
+        default='PALANTIR_ENCLAVE_AUTHENTICATION_BEARER_TOKEN',
+        help='Name of the environment variable holding the auth token you want to use')
+
+    parser.add_argument(
+        '-n', '--datasetName',
+        help='Name of enclave dataset you want to download. CSV will be saved to ValueSet-Tools/data/datasets/<name>')
+
+    parser.add_argument(
+        '-i', '--datasetRid',
+        help='RID of enclave dataset you want to download.')
+    parser.add_argument(
+        '-r', '--ref',
+        default='master',
+        help='Should be the branch of the dataset -- I think. Refer to API documentation at '
+                'https://unite.nih.gov/workspace/documentation/developer/api/catalog/services/CatalogService/endpoints/getTransaction')
+    parser.add_argument(
+        '-f', '--favorites',
+        default=False, action='store_true',
+        help='Just download all the datasets that are currently hardcoded as "favorites".')
+    parser.add_argument(
```

Comment:
New param `--transforms-only`. Can work in combination with `--favorites`.

Reply:
@joeflack4, with --transforms-only, will transforms happen on files that have already been transformed? or are the originals saved and transforms happen on those?

Reply:
@Sigfried I have it reading only from the 'raw' download directory, and writing to the 'repaired' directory. Btw I think it would be nice to rename these directories like `raw/` and `transformed/` or something. Or perhaps the transformed ones can just go directly into `datasets/`.